src/components/code-editor/index.js

```diff
@@ -104,53 +106,70 @@ module.exports = {
 		    modes = ["yaml"];
 		    mimeType = "text/x-yaml";
 	    }
-	    var readOnly = ! this.file.isWritable();
+	    var readOnly = ! this.currentFile.isWritable();
 
-	    this.file.getInputStream(this.context.network, this.context.crypto, props.sizeHigh(), props.sizeLow(), function(read){})
+	    this.currentFile.getInputStream(this.context.network, this.context.crypto, props.sizeHigh(), props.sizeLow(), function(read){})
 	        .thenCompose(function(reader) {
                 var size = that.getFileSize(props);
                 var data = convertToByteArray(new Int8Array(size));
                 return reader.readIntoArray(data, 0, data.length)
                     .thenApply(function(read){
-                        iframe.contentWindow.postMessage({modes:modes, mime:mimeType, readOnly:readOnly, text:new TextDecoder().decode(data)}, '*');
+                        setTimeout(function(){
```

Comment:
why was the settimeout needed?

Reply:
Yes. I was occasionally seeing an empty iframe. This seemed to ensure the iframe got the content.

Reply:
iframe was null?

Reply:
no. felt like a race condition

Reply:
An empty iframe or an empty document? Could this be a fix for https://github.com/Peergos/Peergos/issues/596 ?

Reply:
I am hoping it fixes #596